dart/lib/src/scope.dart

```diff
@@ -153,6 +171,17 @@ class Scope {
       level: level ?? event.level,
     );
 
+    _contexts.clone().forEach((key, value) {
+      // add the runtime list to the event.contexts.runtimes
+      if (key == Runtime.listType && value is List && value.isNotEmpty) {
+        value.forEach((runtime) => event.contexts.addRuntime(runtime));
+      } else if ((!event.contexts.containsKey(key) ||
+              event.contexts[key] == null) &&
+          value != null) {
+        event.contexts[key] = value;
+      }
```

Comment:
why do we need to special case here? its not just adding all the keys from a map to another map?

Reply:
should we override the event.contexts.runtimes ?

Reply:
In Java, it's not a list so it makes sense to override but with a list, I wasn't sure.

Reply:
we should probably merge them, yeah.
can you do a method like you do with `_mergeEventExtra` or `_mergeEventExtra`? seems easier to read.

also, we dont allow null keys right, so `event.contexts[key] == null` seems unecessary?

Reply:
✅

Reply:
for now the `Contexts` ctor allows null values 
```dart
Contexts({
    Device device,
    OperatingSystem operatingSystem,
    List<Runtime> runtimes,
    App app,
    Browser browser,
    Gpu gpu,
  }) : super({
          Device.type: device,
          OperatingSystem.type: operatingSystem,
          Runtime.listType: runtimes ?? [],
          App.type: app,
          Browser.type: browser,
          Gpu.type: gpu,
        });
```

Reply:
true, good catch